core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,372 @@
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToLong(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.longValue();
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue.isFloatingPointNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isFloatingPointNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        BigDecimal retDecimal;
+        try {
+          retDecimal = new BigDecimal(defaultValue.textValue());
+        } catch (NumberFormatException e) {
+          throw new IllegalArgumentException(String.format(
+              "Cannot parse default as a %s value: %s",
+              type, defaultValue), e);
+        }
+        Preconditions.checkArgument(
+            retDecimal.scale() == ((Types.DecimalType) type).scale(),
+            "Cannot parse default as a %s value: %s, the scale doesn't match", type, defaultValue);
+        return retDecimal;
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual() && defaultValue.textValue().length() == 36,
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        UUID uuid;
+        try {
+          uuid = UUID.fromString(defaultValue.textValue());
+        } catch (IllegalArgumentException e) {
+          throw new IllegalArgumentException(String.format("Cannot parse default as a %s value: %s", type,
+              defaultValue), e);
+        }
+        return uuid;
+      case DATE:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return DateTimeUtil.isoDateToDays(defaultValue.textValue());
+      case TIME:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return DateTimeUtil.isoTimeToMicros(defaultValue.textValue());
+      case TIMESTAMP:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          String timestampTz = defaultValue.textValue();
+          Preconditions.checkArgument(DateTimeUtil.isUTCTimestamptz(timestampTz),
+              "Cannot parse default as a %s value: %s, timezone must be UTC",
```

Comment:
This isn't quite correct. The supported timestamp strings don't use zones because they can change. Instead, Iceberg uses offsets. This should be `offset must be +00:00`.

Reply:
Done